File: practice27/10.cpp


//https://www.nowcoder.com/practice/7cf889cb5db44e46a2fcc21c415c0150?tpId=98&tqId=32994&tPage=9&rp=9&ru=/ta/2019test&qru=/ta/2019test/question-ranking
#include<iostream>
#include<string>
#include<vector>
using namespace std;

int main() {
	string str;
	char ch1, ch2;
	while (cin >> str) {
		ch1 = str[0];
		for (int i = 0; i < str.size(); i++) {
			if (str[i] == ',') {
				continue;
			}
			ch2 = str[i];
			if (ch2 > ch1&&ch2 != ch1 + 1) {
				cout << ch1 - '0' + 1 << endl;
				break;
			}
			ch1 = ch2;
		}
	}

	system("pause");
	return 0;
}

File: practice30/10.java



//https://www.nowcoder.com/practice/3e9d7d22b7dd4daab695b795d243315b?tpId=90&tqId=30844&tPage=4&rp=4&ru=/ta/2018test&qru=/ta/2018test/question-ranking
import java.util.*;
public class Main {
    public static int Min_divisor(int m,int n){
        int r=m%n;
        while(r>0){
            m=n;
            n=r;
            r=m%n;
        }
        return n;
    }
    public static void main(String args[]){
        Scanner scanner=new Scanner(System.in);
        int[] pt=new int[5];
        for(int i=0;i<5;i++){
            pt[i]=scanner.nextInt();
        }
        Arrays.sort(pt);
        int Min=Integer.MAX_VALUE;
        int m1,m2;
        for(int i=0;i<3;i++){
            for(int j=i+1;j<4;j++){
                for(int k=j+1;k<5;k++){
                    m1=pt[i]/Min_divisor(pt[i],pt[j])*pt[j];
                    m2=m1/Min_divisor(m1,pt[k])*pt[k];
                    Min=Math.min(Min,m2);
                }
            }
        }
        System.out.println(Min);
    }
}